### ISSUE TITLE
Is `index` inherited or inserted?

### ISSUE THREAD

**Issue by lrhn** [dart-lang/language#1939]
The existing `enum`s have an `index` getter.

The new enhanced enums allow you to apply mixins to the superclass of enums (which is described as being treated as being the `Enum` class, whether it actually is or not) and then introducing an `final int index;` getter in the class itself.

However, the current *implementation* has an internal helper super-class, `_Enum`, which declares `final int index;` and initializes it in a constructor.

If we want to keep doing something like that, the mixins would have to be applied to `_Enum`, which means that they can *override* the inherited `index` variable, where the current proposal specification states that the introduced `index` variable should override the mixins.

What should we do about that?
* Specify that `index` *is* inherited from an unknown super-class, and therefore overridable.
* Make the `index` of `_Enum` private, and add an `int get index => (dart:core::)_index;` getter on each actual `enum` class? (Which can be shared through a mixin too, so it can at least share code, but that adds one more mixin to every `enum` class).
* Something else?

Basically, should we consider changing the proposal spec to support our current implementation, or should we keep it as-is.

(To actually use `Enum` as *superclass* of an `enum` class, it would need a const constructor. We can make it privately named to hide it if needed.)

**Comment by lrhn** [comment]
Index is inherited.
